listenbrainz/webserver/views/playlist_api.py

```diff
@@ -510,20 +510,23 @@ def get_playlist_xspf(playlist_mbid):
     """
 
     if not is_valid_uuid(playlist_mbid):
-        log_raise_400("Provided playlist ID is invalid.")
+        raise PlaylistAPIXMLError("Provided playlist ID is invalid.", status_code=400)
 
     fetch_metadata = parse_boolean_arg("fetch_metadata", True)
-
     playlist = db_playlist.get_by_mbid(playlist_mbid, True)
+
     if playlist is None:
-        raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
+        raise PlaylistAPIXMLError("Cannot find playlist: %s" % playlist_mbid, status_code=404)
+
+    try:
+        user = validate_auth_header(optional=True)
+    except Exception as e:
```

Comment:
You can catch the exact exceptions that `validate_auth_header` raises here instead of a broad `Exception` clause.

Reply:
Thanks for the suggestion.